tharsis/resource_managed_identity_access_rules.go

```diff
@@ -0,0 +1,328 @@
+	}
+
+	// Update the access rule via Tharsis.
+	// The ID is used to find the record to update.
+	// The other fields are modified.
+	updated, err := t.client.ManagedIdentity.UpdateManagedIdentityAccessRule(ctx,
+		&ttypes.UpdateManagedIdentityAccessRuleInput{
+			ID:                     state.ID.ValueString(),
+			RunStage:               ttypes.JobType(plan.RunStage.ValueString()),
+			AllowedUsers:           valueStrings(plan.AllowedUsers),
+			AllowedServiceAccounts: valueStrings(plan.AllowedServiceAccounts),
+			AllowedTeams:           valueStrings(plan.AllowedTeams),
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating managed identity access rule",
+			err.Error(),
+		)
+		return
+	}
+
+	// Copy fields returned by Tharsis to the plan.  Apparently, must copy all fields, not just the computed fields.
+	plan.ID = types.StringValue(updated.Metadata.ID)
+	plan.RunStage = types.StringValue(string(updated.RunStage))
+	plan.ManagedIdentityID = types.StringValue(updated.ManagedIdentityID)
+
+	plan.AllowedUsers = []types.String{}
+	for _, user := range updated.AllowedUsers {
+		plan.AllowedUsers = append(plan.AllowedUsers, types.StringValue(user.Username))
+	}
+
+	plan.AllowedServiceAccounts = []types.String{}
+	for _, serviceAccount := range updated.AllowedServiceAccounts {
+		plan.AllowedServiceAccounts = append(plan.AllowedServiceAccounts, types.StringValue(serviceAccount.ResourcePath))
+	}
+
+	plan.AllowedTeams = []types.String{}
+	for _, team := range updated.AllowedTeams {
+		plan.AllowedTeams = append(plan.AllowedTeams, types.StringValue(team.Name))
+	}
+
+	// Set the response state to the fully-populated plan.
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (t *managedIdentityAccessRuleResource) Delete(ctx context.Context,
+	req resource.DeleteRequest, resp *resource.DeleteResponse) {
+
+	// Get the current state.
+	var state ManagedIdentityAccessRuleModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Delete the managed identity via Tharsis.
+	// The ID is used to find the record to delete.
+	err := t.client.ManagedIdentity.DeleteManagedIdentityAccessRule(ctx,
+		&ttypes.DeleteManagedIdentityAccessRuleInput{
+			ID: state.ID.ValueString(),
+		})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error deleting managed identity access rule",
+			err.Error(),
+		)
+		return
+	}
+}
+
+// ImportState helps the provider implement the ResourceWithImportState interface.
+func (t *managedIdentityAccessRuleResource) ImportState(ctx context.Context,
+	req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
+
+	// Retrieve import ID and save to id attribute
+	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
+}
+
+// valueStrings converts a slice of types.String to a slice of strings.
+func valueStrings(arg []types.String) []string {
```

Comment:
This function can be added as a method to the `managedIdentityAccessRuleResource` struct.